app/services/paypal_service.rb

```diff
@@ -7,7 +7,22 @@
 )
 
 class PaypalService
-  def create_agreement(id)
+  def initialize(plan)
+    @plan = plan
+  end
+
+  def create_and_activate_recurring_plan
+    plan = if  @plan.third_party_identifier === 'premium'
+             PayPal::SDK::REST::Plan.new(plan_params_with_trial)
+           else
+             PayPal::SDK::REST::Plan.new(plan_params_without_trial)
+           end
```

Comment:
alternatively could be 

```rb
plan = PayPal::SDK::REST::Plan.new(@plan.third_party_identifier === 'premium' ? plan_params_with_trial : plan_params_without_trial)
```

or

```rb
plan_type = @plan.third_party_identifier === 'premium' ? plan_params_with_trial : plan_params_without_trial
plan = PayPal::SDK::REST::Plan.new(play_type)
```

which make the method a little shorter, but let me know what you think about whether it's actually an improvement in readability?